pandas/_libs/tslibs/strptime.pyx

```diff
@@ -134,7 +134,7 @@ def array_strptime(ndarray[object] values, object fmt, bint exact=True, errors='
                 iresult[i] = NPY_NAT
                 continue
         else:
-            if checknull_with_nat(val):
+            if checknull_with_nat(val) or val.__repr__() == "<NA>":
```

Comment:
May be it's better to write this check `val.__repr__() == "<NA>"` in function `checknull_with_nat`?

Reply:
this is non-performant, we use instance checks in cython